engine/dlib/src/dlib/sys_cocoa.mm

```diff
@@ -163,16 +163,11 @@ Result GetLogPath(char* path, uint32_t path_len)
 
     Result OpenURL(const char* url, const char* target)
     {
-        NSString* ns_url = [NSString stringWithUTF8String: url];
-        BOOL ret = [[UIApplication sharedApplication] openURL:[NSURL URLWithString: ns_url]];
-        if (ret == YES)
-        {
-            return RESULT_OK;
-        }
-        else
-        {
-            return RESULT_UNKNOWN;
-        }
+        NSString* ns_str = [NSString stringWithUTF8String:url];
+        NSURL* ns_url = [NSURL URLWithString:ns_str];
+        UIApplication *app = [UIApplication sharedApplication];
+        [app openURL:ns_url options:@{} completionHandler:nil];
+        return [app canOpenURL:ns_url] ? RESULT_OK : RESULT_UNKNOWN;
```

Comment:
The new `openURL` function is asynchronous, so we can't get a result the same way we did before, but we can use another function to check if URL can be opened or not, and return this info instead.